underdocs-error/src/test/kotlin/underdocs/error/pretty/DefaultErrorPrettyPrinterTest.kt

```diff
@@ -0,0 +1,64 @@
+package underdocs.error.pretty
+
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+import underdocs.error.BaseUnderdocsError
+
+class DefaultErrorPrettyPrinterTest {
+
+    companion object {
+        const val TITLE = "Test Exception"
+        const val ERROR_CODE = "42"
+        const val DESCRIPTION = "This is a dummy exception for tests."
+        const val SOLUTION = "There is no solution for this."
+        const val RUNTIME_EXCEPTION_CLASSPATH = "java.lang.RuntimeException"
+        const val RUNTIME_EXCEPTION_DESCRIPTION = "RuntimeException Description"
+    }
+
+    private val defaultErrorPrettyPrinter = DefaultErrorPrettyPrinter()
+
+    @Test
+    fun print_WhenBaseUnderDocsErrorTypeExceptionIsThrown_ShouldReturnMessageContainingTheDetailsOfTheException() {
+        // Given
+        val expectedBaseUnderDocsErrorTypeMessage =
+                """
+                |${TITLE.toUpperCase()} - Error Code $ERROR_CODE
+                |Details
+                |${DESCRIPTION}
+                |
+                |Possible solution
+                |${SOLUTION}
+            """.trimMargin()
+
+        // When
+        val message = defaultErrorPrettyPrinter.print(TestException(ERROR_CODE, TITLE))
+
+        // Then
+        assertEquals(expectedBaseUnderDocsErrorTypeMessage, message)
+    }
+
+    @Test
+    fun print_WhenUnexpectedExceptionIsThrown_ShouldReturnMessageContainingTheDetailsOfTheException() {
```

Comment:
I see, you're using some kind of a prefix to group test cases exercising the same function (in this case, `print`). This is a good practice, which we can even take further using JUnit5.

Consider the following structure:

~~~~Kotlin
@DisplayName("In DefaultErrorPrettyPrinter")
class DefaultErrorPrettyPrinterTest {
  @Nested
  @DisplayName("the function print")
  inner class Print {
    @Test
    fun `should return message containing the details of the exception when BaseUnderdocsError type exception is thrown.`() {
      // Test code.
    }
  }
}
~~~~

Keep in mind, that we're walking all these extra miles to get a nice error report. Let me break the above snippet down, line by line:

  * Applied a class-level `@DisplayName` annotation to get rid of the `Test` postfix in the test report and make it read like the beginning of a sentence.
  * Added the nested modifier to an inner class so that JUnit would analyze that inner class as well. See https://junit.org/junit5/docs/current/user-guide/#writing-tests-nested
  * Here, I applied the `@DisplayName` annotation again, to continue the sentence started by the previous annotation.
  * The class is declared as an `inner` class, so that it can access the members of the enclosing class. See https://kotlinlang.org/docs/reference/nested-classes.html#inner-classes
  * I exchanged the order of the `Should` and `When` clauses in the test name to make it read better and lower-cased them. Also, closed the sentence with a full stop (or period).

The result is the following error report by JUnit5:

~~~~
* Test Results
  * In DefaultErrorPrettyPrinter
    * the function print
      * should return message containing the details of the exception when BaseUnderdocsError type exception is thrown.
~~~~

Note, that it reads perfectly well and is nested according the grouping we wanted to achieve (group-by-function).

Reply:
It's brilliant! Everything works great and looks nice!
However, the full stop doesn't compile, so I had to remove it.

Reply:
Nice catch! Adding the full stop was just an afterthought that I did not bother trying out in the IDE. I'm quite astonished, it causes a compilation error though :D